template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Disable2fa.cshtml.cs

```diff
@@ -28,12 +28,12 @@ public class Disable2faModel : PageModel
             var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with name '{User.Identity.Name}'.");
+                throw new ApplicationException($"Unable to load user with id '{_userManager.GetUserId(User)}'.");
```

Comment:
I don't have both the `UserManager` and `SignInManager` in every page, so that seems like overkill just to make this error message. So far, the error occurs when I have a stale cookie after nuking the database, and the app would have to change to get into a situation where you had a valid user principal that didn't contain the ID (e.g. add Windows Auth).

We need to make a call here and move on.